src/main/java/org/quiltmc/loader/impl/plugin/QuiltPluginManagerImpl.java

```diff
@@ -1921,26 +1921,17 @@ private void addModOption(Path file, Map<ModLoadOption, BasePluginContext> map,
 	}
 
 	void addSingleModOption(ModLoadOption mod, BasePluginContext provider, boolean only, PluginGuiTreeNode guiNode) {
-
-		PluginGuiTreeNode loadedBy = guiNode.addChild(QuiltLoaderText.translate("gui.text.mod_loaded_by", provider.pluginId()))//
-			.mainIcon(mod.modTypeIcon());
-
-		if (only) {
-			guiNode.subIcon(mod.modTypeIcon());
-			loadedBy.debug();
-		}
-
 		String id = mod.id();
 		Version version = mod.version();
 		Path from = mod.from();
-		modPaths.put(from, mod);
-		modProviders.put(mod, provider.pluginId());
-		modGuiNodes.put(mod, guiNode);
-
-		guiNode.addChild(QuiltLoaderText.translate("gui.text.id", id));
-		guiNode.addChild(QuiltLoaderText.translate("gui.text.version", version.raw()));
 
 		PotentialModSet set = modIds.computeIfAbsent(id, k -> new PotentialModSet());
+
+		if (set.byVersionSingles.containsKey(version) && QuiltLoader.isDevelopmentEnvironment()) {
+			Log.debug(LogCategory.SOLVING, String.format("Ignoring duplicate mod %s of the same version %s loaded from %s", id, version, from));
```

Comment:
```suggestion
			Log.warn(LogCategory.SOLVING, String.format("Ignoring duplicate mod %s of the same version %s loaded from %s", id, version, from));
```
That would make it a noisy log error - I don't think `debug` shows by default.